pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,9 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration != pdb.Generation {
```

Comment:
This is a good point. I though that Generation is automatically incremented by our infrastructure on every Spec update, but it seems this has to be done manually. For example RC is doing it here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/controller/strategy.go#L81

@mwielgus - this should be added.